src/components/organisms/SpaceTimingForm/SpaceTimingForm.tsx

```diff
@@ -0,0 +1,219 @@
+
+import { WithId } from "utils/id";
+import {
+  convertDateFromUtcSeconds,
+  convertUtcSecondsFromInputDateAndTime,
+} from "utils/time";
+
+import { useUser } from "hooks/useUser";
+
+import { AdminSidebarFooterProps } from "components/organisms/AdminVenueView/components/AdminSidebarFooter/AdminSidebarFooter";
+
+import { AdminDateTime } from "components/molecules/AdminDateTime";
+import { AdminSection } from "components/molecules/AdminSection";
+import { FormErrors } from "components/molecules/FormErrors";
+import { LoadingPage } from "components/molecules/LoadingPage";
+import { SubmitError } from "components/molecules/SubmitError";
+
+import { ButtonNG } from "components/atoms/ButtonNG";
+
+import "./SpaceTimingForm.scss";
+
+// NOTE: add the keys of those errors that their respective fields have handled
+const HANDLED_ERRORS: string[] = [
+  "start",
+  "startDate",
+  "startTime",
+  "end",
+  "endDate",
+  "endTime",
+];
+
+export interface SpaceTimingFormInput {
+  startDate: string;
+  startTime: string;
+  endDate: string;
+  endTime: string;
+}
+
+const testEndDate: TestFunction = function testEndIsAfterStart() {
+  const { startDate, endDate } = this.parent;
+
+  const zeroHour = "00:00";
+  const start = convertUtcSecondsFromInputDateAndTime({
+    date: startDate,
+    time: zeroHour,
+  });
+  const end = convertUtcSecondsFromInputDateAndTime({
+    date: endDate,
+    time: zeroHour,
+  });
+
+  return start <= end;
+};
+
+const testEndTime: TestFunction = function testEndIsAfterStart() {
+  const { endTime, startTime, startDate, endDate } = this.parent;
+
+  const start = convertUtcSecondsFromInputDateAndTime({
+    date: startDate,
+    time: startTime,
+  });
+  const end = convertUtcSecondsFromInputDateAndTime({
+    date: endDate,
+    time: endTime,
+  });
+
+  return start < end;
+};
+
+const validationSchema = Yup.object().shape({
+  startDate: Yup.string()
+    .required()
+    .test(
+      "startDate",
+      "Start date must not be after the end date",
+      testEndDate
+    ),
+  startTime: Yup.string()
+    .required()
+    .test("startTime", "Start time must be before the end time", testEndTime),
+  endDate: Yup.string()
+    .required()
+    .test(
+      "endDate",
+      "End date must not be before the starting date",
+      testEndDate
+    ),
+  endTime: Yup.string()
+    .required()
+    .test("endTime", "End time must be after the start time", testEndTime),
+});
+
+export interface SpaceTimingFormProps extends AdminSidebarFooterProps {
+  venue?: WithId<AnyVenue>;
+}
+
+export const SpaceTimingForm: React.FC<SpaceTimingFormProps> = ({ venue }) => {
+  const { user } = useUser();
+
+  const defaultValues = useMemo<SpaceTimingFormInput>(() => {
+    const {
+      inputFormattedDateSegment: startDate,
+      inputFormattedTimeSegment: startTime,
+    } = convertDateFromUtcSeconds(venue?.start_utc_seconds ?? NaN);
+
+    const {
+      inputFormattedDateSegment: endDate,
```

Comment:
I wanted to go with long overly descriptive names just because of the nature of having many date/time formats (and  the possibility of expanding that result)

Reply:
In this case they're using `_INPUT_` while in other case the constant may be `TIME_AND_DATE` which may or may not be related to how `<input>`s work.